rabbit-escape-render/src/rabbitescape/render/gameloop/GeneralPhysics.java

```diff
@@ -104,6 +124,25 @@ public boolean gameRunning()
         return ( world.completionState() == World.CompletionState.RUNNING );
     }
 
+    /**
+     * Take actions for demo mode, eg drop tokens.
+     */
+    private void doInterpreterActions()
+    {
+        SolutionTimeStep stp = solutionInterpreter.next( CompletionState.RUNNING );
+        for ( TimeStepAction action: stp.actions )
+        {
+            if ( action instanceof SelectAction )
+            {
+                actionTypeSwitch.caseSelectAction( (SelectAction)action );
+            }
+            else if ( action instanceof PlaceTokenAction )
+            {
+                actionTypeSwitch.casePlaceTokenAction( (PlaceTokenAction)action );
+            }
```

Comment:
>  I was expecting these type switch classes to only be implemented inline.

I don't understand. Sorry, you will have to spell it out for me.

GeneralPhysics needs to make calls back to swing, but render is not allowed to depend on swing, so I need an interface in render (or engine) that swing can implement.